File: SortVision/public/code/selection/ruby/selectionSort.rb
# Selection Sort (in-place)
# Time: Best/Avg/Worst O(n^2), Space: O(1)
def selection_sort(arr)
  (0...arr.length - 1).each do |i|
    min_idx = i
    (i + 1...arr.length).each do |j|
      min_idx = j if arr[j] < arr[min_idx]
    end
    arr[i], arr[min_idx] = arr[min_idx], arr[i] if min_idx != i
  end
  arr
end

arr = [64, 25, 12, 22, 11]
p selection_sort(arr)

File: SortVision/public/code/selection/php/selectionSort.php
<?php
// Selection Sort (in-place)
// Time: Best/Avg/Worst O(n^2), Space: O(1)
function selectionSort(array &$arr): void {
    $n = count($arr);
    for ($i = 0; $i < $n - 1; $i++) {
        $minIdx = $i;
        for ($j = $i + 1; $j < $n; $j++) {
            if ($arr[$j] < $arr[$minIdx]) $minIdx = $j;
        }
        if ($minIdx !== $i) {
            [$arr[$i], $arr[$minIdx]] = [$arr[$minIdx], $arr[$i]];
        }
    }
}

$arr = [64, 25, 12, 22, 11];
selectionSort($arr);
echo implode(', ', $arr) . PHP_EOL;
?>
